lib/elixir/lib/kernel/parallel_compiler.ex

```diff
@@ -649,19 +652,51 @@ defmodule Kernel.ParallelCompiler do
           state
         )
 
-      {:module_available, child, ref, file, module, binary} ->
+      {:load_module?, child, ref, module} ->
+        # If compiling files to disk, we only load the module
+        # if other modules are waiting for it.
```

Comment:
```suggestion
        # if other modules are waiting for it. This is done as we prefer
        # to load the module in the same process that defines it,
        # to avoid extra copies and to avoid blocking the parallel compiler.
```